core/src/main/java/org/elasticsearch/index/engine/ShadowEngine.java

```diff
@@ -245,4 +245,9 @@ protected SegmentInfos getLastCommittedSegmentInfos() {
         return lastCommittedSegmentInfos;
     }
 
+    @Override
+    public long indexWriterRAMBytesUsed() {
+        // No IndexWriter
+        return 0L;
```

Comment:
I wonder if UnsupportedOperationException is a better choice here- we don't have an indexwriter...